### PR TITLE
Update EIP-4750: Add reference implementation

### PR DESCRIPTION
**Pull request by gumb0**

### CODE REVIEW

File: EIPS/eip-4750.md [hunk 1]
```diff
@@ -133,6 +133,171 @@ Alternative logic for executing `RETF` in the top frame could be to exceptionall
 
 We have decided in favor of always having at least one item in the return stack, because it allows to avoid having a special case for empty stack in the interpreter loop stack underflow check. We keep the stack underflow rule general by having `caller_stack_height = 0` in the top frame.
 
+## Reference Implementation
```
Comment (xinbenlv) [inline EIPS/eip-4750.md:136]: Thank you, it's super valuable to include the reference implementation. Could you move it after the backward compatbiility?

File: EIPS/eip-4750.md [hunk 2]
```diff
@@ -133,6 +133,171 @@ Alternative logic for executing `RETF` in the top frame could be to exceptionall
...
+    immediate_sizes[opcode] = opcode - 0x60 + 1
+
+class ValidationException(Exception):
+    pass
+
+# Validate EOF code.
+# Raises ValidationException on invalid code
+def validate_eof(code: bytes):
+    # Check version
+    if len(code) < 3 or code[2] != VERSION:
```
Comment (xinbenlv) [inline EIPS/eip-4750.md:181]:
```suggestion
    assert code[2] == VERSION, "Version must match 0x01"
```

File: EIPS/eip-4750.md [hunk 3]
```diff
@@ -133,6 +133,171 @@ Alternative logic for executing `RETF` in the top frame could be to exceptionall
...
+        raise ValidationException("invalid version")
```
Comment (xinbenlv) [inline EIPS/eip-4750.md:182]:
```suggestion
```

File: EIPS/eip-4750.md (hunk 2 above)
Comment (xinbenlv) [reply EIPS/eip-4750.md:181]: Consider simplifying assert

Comment (gumb0) [reply EIPS/eip-4750.md:187]: See https://github.com/ethereum/EIPs/pull/5663#discussion_r976771485 for reasoning to use `raise`

**Pull request merged**